File: BLH.ApproveIQ.Domain/Entities/InvoiceItem.cs
using BLH.ApproveIQ.Domain.Primitives;

namespace BLH.ApproveIQ.Domain.Entities;

public class InvoiceItem : AuditableEntity
{
    // Foreign Key to Invoice
    public Guid InvoiceId { get; set; }

    // Columns as shown in the "Invoice Line" table
    public string? BusinessUnit { get; set; }            // "Business Unit"
    public string? BuDescription { get; set; }           // "BU Description"
    public string? CostCode { get; set; }                // "Cost Code"
    public string? CostCodeDesc { get; set; }            // "Cost Code Desc"
    public string? CostType { get; set; }                // "Cost Type"
    public string? CostTypeDesc { get; set; }            // "Cost Type Desc"
    public string? OrderNumber { get; set; }             // "Order Number"
    public string? Line { get; set; }                    // "Line"
    public string? OrderSuffix { get; set; }             // "Order Suf..."
    public string? GlLineType { get; set; }              // "GL Line Type"
    public decimal? Qty { get; set; }                    // "QTY"
    public string? Uom { get; set; }                     // "UOM"
    public decimal? UnitPrice { get; set; }              // "Unit Price"
    public decimal? Amount { get; set; }                 // "Amount"

    // Navigation Property
    public Invoice Invoice { get; set; } = null!;
}

File: BLH.ApproveIQ.Portal/ClientApp/src/app/data/data/b-l-h/entities/invoice-item.ts
import { AuditableEntity } from '../../../b-l-h/primitives/auditable-entity';
import { Invoice } from './invoice';

export class InvoiceItem extends AuditableEntity {
    invoiceId: string;
    businessUnit: string;
    buDescription: string;
    costCode: string;
    costCodeDesc: string;
    costType: string;
    costTypeDesc: string;
    orderNumber: string;
    line: string;
    orderSuffix: string;
    glLineType: string;
    qty?: number;
    uom: string;
    unitPrice?: number;
    amount?: number;
    invoice = new Invoice();
}
